mlir/lib/Dialect/XeGPU/IR/XeGPUOps.cpp

```diff
@@ -121,6 +121,39 @@ isValidGatherScatterParams(Type maskTy, VectorType valueTy,
   return success();
 }
 
+// Verify that number of offsets matches either the source rank or the tdesc
```

Comment:
I think we should simply check that offsets can't have lower rank than tdesc rank. It shoud be fine if it is larger than tdesc rank. 

I am not sure that it is common practice for the op validation to validate itself using information from producer op. I would rather checking this in transformation or lowering passes.  Any opinion from @adam-smnk ?

Reply:
Op verifier should be contained to the op itself without accessing any external data.

Reply:
> I think we should simply check that offsets can't have lower rank than tdesc rank. It shoud be fine if it is larger than tdesc rank.

Sound reasonable. Just please clearly define how offsets are interpreted.
For example, memref is 4D, descriptor is 2D and there are 3 offsets. I guess they'd apply to the innermost dimensions.

Reply:
simplified the check to not rely on `CreateNdTdescOp`.

Also added docs for the new offset syntax